src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -214,6 +214,46 @@ public void AddTagHelperAttribute(string name, object value)
             _allAttributes.Add(name, value);
         }
 
+        // Used to pool TagHelperExecutionContext's. With this we don't always need to recreate TagHelperExecutionContexts.
+        internal void ClearAndUpdateWith(
```

Comment:
Nope, user code wont see this type. As for `Reset`, that was my first naming of this but felt that the name mislead with what it really did.